config/develop/example-dev-project.j2

```diff
@@ -1,28 +1,40 @@
-stack_name: example-dev-project
+{% set stack_name = "example-dev-project" %}
+{% set read_write_users = ["bruno.grande@sagebase.org"] %}
+{% set read_only_users = ["tess.thyer@sagebase.org", "thomas.yu@sagebase.org"] %}
 
-stack_tags:
-  Department: IBC
-  Project: Infrastructure
-  OwnerEmail: nextflow-admins@sagebase.org
+# DO NOT EDIT ANYTHING BELOW THIS LINE -----------------------------------------------------------
+
+template_path: tower-project.yaml
+stack_name: {{ stack_name }}
+dependencies:
+  - common/nextflow-forge-iam-policy.yaml
+  - common/nextflow-launch-iam-policy.yaml
 
 parameters:
+  {% if read_write_users %}
   S3ReadWriteAccessArns:
-    - arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/bruno.grande@sagebase.org
+  {% for user in read_write_users %}
+    - arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/{{ user }}
+  {% endfor %}
+  {% endif %}
+  {% if read_only_users %}
   S3ReadOnlyAccessArns:
-    - arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/tess.thyer@sagebase.org
-
-  # DO NOT EDIT ANYTHING BELOW THIS LINE ------------------------------------------------------------------------
-
+  {% for user in read_only_users %}
+    - arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/{{ user }}
```

Comment:
I'm guessing the vast majority of Tower users will have access to the Developer role. In the rare event that they don't, we can edit the template to use Viewer instead. 

Otherwise, this template will jump in complexity. One workaround is to preemptively list both the Developer and Viewer roles for all emails in `read_only_users`, but this feels hacky. 

@tthyer: What do you think?

Reply:
@BrunoGrandePhD I assumed the opposite. The only people who need Developer are those who are working on the infrastructure. Everyone else is a Tower User/Viewer. They need to be able to view things in the accounts but otherwise they're just going to be running workflows.

Reply:
The users listed under `S3ReadWriteAccessArns` need `AmazonS3FullAccess`, which only comes with the Developer role and not the Viewer role. Otherwise, they won't be able to write to the S3 bucket given that cross-account access only works if policies in both accounts include explicit allows. 

In any case, because this relies on cross-account access, it doesn't matter what the Developer role grants; the only resources that they can access are those where they're listed explicitly in our account. So, I'm not afraid of accidentally granting them access to more than is needed (in this case, the KMS key and the S3 bucket). 

Another argument for settling on the sandbox Developer role is that someone who's initially read-only on one Tower project could eventually be added as read/write on another. It would reduce the number of IT JIRA tickets if we just expected Tower users to ask for the sandbox Developer role.

Reply:
Ok, I was reading it as Tower devs, not Sandbox devs, but now that you've reminded me about the cross-account access, it makes sense.